### PR TITLE
fix routing for custom export

### PR DESCRIPTION
**Pull request by oallain**
If we use custom exporter this "Backup route" catch page call and make an fatal error.

So i fixed by using spéficic route for each export.

This not a BC, because app_export_data_*  respect the format used and described in the doc.

### CODE REVIEW

**Comment by stefandoorn** [comment]
Build is breaking, seems there are resources without a dot in it..

**Comment by oallain** [comment]
Yes, i had fixed it :smile: 

Reading the code, I also realized that the spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php, was not correct.

File: src/Listener/ExportButtonGridListener.php [hunk 1]
```diff
@@ -89,10 +89,17 @@ private function addLink(string $format): array
             $parameters['criteria'] = $currentRequest->query->get('criteria');
         }
 
+        $ressource = $this->resource;
+
+        $explode = \explode('.', $this->resource);
```
Comment (lsmith77) [inline src/Listener/ExportButtonGridListener.php]: imho we should not prefix core functions with `\`

Comment (oallain) [reply src/Listener/ExportButtonGridListener.php]: Yes, you are right.
It is not part of the list of functions optimized by opcode ;)
See https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php#L346

Comment (oallain) [reply src/Listener/ExportButtonGridListener.php]:
```suggestion
        $explode = explode('.', $this->resource);
```

File: spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php [hunk 2]
```diff
@@ -39,29 +40,30 @@ function it_processes_the_exporter_services(
...
         ]);
 
         $container->getParameter('sylius.exporter.web_ui')->willReturn(true);
-        $container->has('app.grid_event_listener.admin.crud_csv_exporter_format_export')->willReturn(false);
+        $container->has('app.grid_event_listener.admin.crud_' . $exporterType . '_' . $exporterFormat . '_export')->willReturn(false);
+        $container->has('sylius.controller.export_data_' . $exporterType)->willReturn(false);
 
         $container->register(
-            'app.grid_event_listener.admin.crud_csv_exporter_format_export',
+            'app.grid_event_listener.admin.crud_' . $exporterType . '_' . $exporterFormat . '_export',
```
Comment (bitbager) [inline spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php]: You can use \sprintf('app.grid_event_listener.admin.crud_%s_%s_export', $exporterType, $exporterFormat) as a shorthand :)

File: spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php [hunk 3]
```diff
@@ -39,29 +40,30 @@ function it_processes_the_exporter_services(
...
-        $blockEventDefinition->addArgument('csv')->willReturn($blockEventDefinition);
+        $blockEventDefinition->addArgument($exporterType)->willReturn($blockEventDefinition);
+        $blockEventDefinition->addArgument([$exporterFormat])->willReturn($blockEventDefinition);
         $blockEventDefinition->addMethodCall('setRequest', Argument::that(function ($input) {
             return is_array($input);
         }))->willReturn($blockEventDefinition);
         $blockEventDefinition->addTag('kernel.event_listener',
             [
-                'event' => 'sylius.grid.admin_csv',
+                'event' => 'sylius.grid.admin_' . $exporterType,
```
Comment (bitbager) [inline spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php]: \sprintf('sylius.gird.admin_%s', $exporterType) might work here as well

File: spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php (hunk 2 above)
Comment (oallain) [reply spec/DependencyInjection/Compiler/RegisterExporterPassSpec.php]: Yes, of course.

**Comment by oallain** [comment]
Hello @lsmith77 @stefandoorn @bitbager @solutionDrive-Alt 

This pull request is blocking for me ;)
Do you thinks we can merge it ?

Thanks

**Comment by vasilvestre** [comment]
> LGTM, except for prefixing the php core functions which imho hurts readability

As it allow you to avoid local namespace override and allow some opcode optimisation, prefixing is important :)

**Comment by oallain** [comment]
Thank you guys for reading and approval :smile:

**Pull request merged**